lib/IRGen/StackSizeLimitTransform.cpp

```diff
@@ -0,0 +1,307 @@
+//===--- StackSizeLimitTransform.cpp --------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// This is a transform that ensures that swift allocations never blow out the
+/// stack. We do this by limiting per function the amount of stack space
+/// allocated and otherwise, heap allocate classes, values.
```

Comment:
Please add that this pass also lowers the builtin and therefore is mandatory to run (which is the case for all IRGen passes anyway, but I think it doesn't harm to mention it)

Reply:
Sounds good!